crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -224,19 +225,69 @@ where
         Ok(hash)
     }
 
-    async fn send_transaction(&self, request: TransactionRequest) -> EthResult<H256> {
+    async fn send_transaction(&self, mut request: TransactionRequest) -> EthResult<H256> {
         let from = match request.from {
             Some(from) => from,
             None => return Err(SignError::NoAccount.into()),
         };
+
+        // set nonce if not already set before
+        if request.nonce.is_none() {
+            let nonce =
+                self.get_transaction_count(from, Some(BlockId::Number(BlockNumberOrTag::Pending)))?;
+            request.nonce = Some(nonce.checked_add(U256::from(1)).ok_or(EthApiError::MaxNonce)?);
+        }
+
+        let chain_id = self.chain_id();
+        // we need an oracle to fetch the gas price of the current chain
+        let gas_price = request.gas_price.unwrap_or_default();
+        let max_fee_per_gas = request.max_fee_per_gas.unwrap_or_default();
+
+        let estimated_gas = self
+            .estimate_gas_at(
+                CallRequest {
+                    from: Some(from),
+                    to: request.to,
+                    gas: request.gas,
+                    gas_price: Some(U256::from(gas_price)),
+                    max_fee_per_gas: Some(U256::from(max_fee_per_gas)),
+                    value: request.value,
+                    data: request.data.clone(),
+                    nonce: request.nonce,
+                    chain_id: Some(chain_id),
+                    access_list: request.access_list.clone(),
+                    max_priority_fee_per_gas: Some(U256::from(max_fee_per_gas)),
```

Comment:
I try to run this method and the underlying `ensure_fees` function return error `Err(EthApiError::ConflictingFeeFieldsInRequest)`. I debugged and found that there is no case that all of three `gas_price`, ` max_fee_per_gas` and `  max_priority_fee_per_gas` is Some(...). Could anyone please help me with this point ?

Reply:
could you please open a new issue for this?